### PR TITLE
fix: invoke correct rpc for truncate bcast tree

### PR DESCRIPTION
**Pull request by adammoody**
@sandrain , would you please double check me on this change?

### CODE REVIEW

**Comment by sandrain** [comment]
@adammoody @MichaelBrim Yes, looks like that was a bug. Thanks.

File: server/src/unifyfs_group_rpc.c [hunk 1]
```diff
@@ -715,7 +715,7 @@ int truncate_bcast_forward(const unifyfs_tree_t* broadcast_tree,
                    i, child, glb_servers[child].margo_svr_addr_str);
 
             /* allocate handle */
-            rc = get_request_handle(unifyfsd_rpc_context->rpcs.truncate_id,
+            rc = get_request_handle(unifyfsd_rpc_context->rpcs.truncate_bcast_id,
```
Comment (sandrain) [inline server/src/unifyfs_group_rpc.c:718]: Isn't this line going over 80 (checkpatch may fail)?

**Comment by MichaelBrim** [comment]
Yeah, I think this is something I already caught in the readrpc branch, but forgot to backport.

**Review by MichaelBrim — APPROVED** [review]

**Comment by clmendes** [comment]
Yes, with this fix I was able to run correctly the HDF5 example (from Issue #577) on 2 processors and 2 Catalyst nodes (i.e. 1 processor/node).

However, when I try to run it with 4 processors in 2 nodes (i.e. 2 processors/node) I still get the same error as before. Thus, I'm afraid this might not yet be the last fix.

**Comment by adammoody** [comment]
Thanks @sandrain and @MichaelBrim

**Pull request merged**